### PR TITLE
Jsaal 36

### PR DESCRIPTION
**Pull request by Aaron-Xiaoxingchen**
Can you help me check the set up environment? Thanks

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,7 @@
+node_modules
+.env
```
Comment (FairyWind) [inline .gitignore:2]:
```suggestion
.env
npm-debug.*
```

**Review by FairyWind — CHANGES_REQUESTED** [review]
You need to build a database and initial it in app.js.
Please add an .env.template file also.
Better to test the server when finishing.

File: .eslintrc.json [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    "airbnb-base"
+  ],
+  "globals": {
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly"
+  },
+  "parserOptions": {
+    "ecmaVersion": 2018
+  },
+  "rules": {}
```
Comment (ikarasz) [inline .eslintrc.json]: we should allow `console.log`

File: App.js [hunk 3]
```diff
@@ -0,0 +1 @@
+console.log('Testing');
```
Comment (ikarasz) [inline App.js]: pls set up a basic express endpoint here

**Review by ikarasz — CHANGES_REQUESTED** [review]

**Review by FairyWind — APPROVED** [review]

File: App.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
```
Comment (ikarasz) [inline App.js]: do not use var :sob:

File: App.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+var app = express();
+
+app.get('/', (req, res) => {
+  res.send('Hello World');
+})
+
+app.listen(3000, () => {
```
Comment (ikarasz) [inline App.js]: pls use environemnt variables

File: App.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+var express = require('express');
+var app = express();
+
+app.get('/', (req, res) => {
+  res.send('Hello World');
+})
```
Comment (ikarasz) [inline App.js]: did you run eslint?

File: package.json [hunk 7]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "jsa-alexandrite-backend",
+  "version": "1.0.0",
+  "description": "",
+  "main": "App.js",
+  "scripts": {
+    "test": "mocha 'tests/**/*.js' --exit",
```
Comment (ikarasz) [inline package.json:7]: write a dummy test as well

File: package.json [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+    "lint": "eslint .",
+    "start": "node ./App.js",
+    "start:dev": "nodemon ./App.js"
```
Comment (ikarasz) [inline package.json]: add nodemon as dependency

File: package.json [hunk 9]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "jsa-alexandrite-backend",
+  "version": "1.0.0",
+  "description": "",
+  "main": "App.js",
+  "scripts": {
+    "test": "mocha 'tests/**/*.js' --exit",
+    "lint": "eslint .",
```
Comment (ikarasz) [inline package.json:8]: this will check the node_modules folder as well

**Review by ikarasz — CHANGES_REQUESTED** [review]
- do not put the app.js in the root
- read the story specification before creating a pull request: https://jira.greenfox.academy/browse/JSAAL-36, database is needed as well

File: src/App.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
+const express = require('express');
+
+const app = express();
+const PORT = 3000;
+
```
Comment (FairyWind) [inline src/App.js:6]: you should use environment variable from .env here

**Review by FairyWind — CHANGES_REQUESTED** [review]
fix the port

**Review by FairyWind — APPROVED** [review]

File: .eslintrc.json [hunk 11]
```diff
@@ -0,0 +1,21 @@
+{
+  "env": {
+      "browser": true,
```
Comment (ikarasz) [inline .eslintrc.json]: do you wanna run the backend in the browser? :)

File: package.json [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+    "start": "node .src//App.js",
```
Comment (ikarasz) [inline package.json]: are you sure this will work?

File: package.json [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+    "url": "git+https://github.com/green-fox-academy/jsa-alexandrite-backend.git"
+  },
+  "author": "",
+  "license": "ISC",
+  "bugs": {
+    "url": "https://github.com/green-fox-academy/jsa-alexandrite-backend/issues"
+  },
+  "homepage": "https://github.com/green-fox-academy/jsa-alexandrite-backend#readme",
+  "dependencies": {
+    "assert": "^2.0.0",
```
Comment (ikarasz) [inline package.json]: is it really a production dependency?

File: package.json [hunk 14]
```diff
@@ -0,0 +1,36 @@
...
+    "dotenv": "^8.2.0",
+    "express": "^4.17.1",
+    "mocha": "^6.2.2",
```
Comment (ikarasz) [inline package.json]: is it really a production dependency?

File: package.json [hunk 15]
```diff
@@ -0,0 +1,36 @@
...
+    "mysql2": "^2.0.2",
+    "nodemon": "^2.0.2",
```
Comment (ikarasz) [inline package.json]: is it really a production dependency?

File: src/App.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
...
+const app = express();
+
+app.use(express.json());
+
+app.get('/', (req, res) => {
+  res.send('Hello World');
+});
+
+const { PORT } = process.env;
```
Comment (ikarasz) [inline src/App.js]: variable declaration should go to the top of the file

**Review by ikarasz — CHANGES_REQUESTED** [review]

**Review by ikarasz — APPROVED** [review]

**Pull request merged**